src/main/java/org/apache/maven/doxia/DefaultConverter.java

```diff
@@ -78,57 +77,160 @@
+        FO( "fo", "fo", false, true ),
+        ITEXT( "itext", "itext", false, true ),
+        LATEX( "latex", "tex", false, true ),
+        TWIKI( "twiki", "twiki", true, true ),
+        RTF( "rtf", "rtf", false, true ),
+        XDOC( "xdoc", "xml", "document", true, true ),
+        XHTML( "xhtml", "html", "html", true, true ),
+        XHTML5( "xhtml5", "html", true, true ), // no autodetect support
+        MARKDOWN( "markdown", "md", false, true );
 
-    private static final String APT_SINK = "apt";
+        /** Plexus role hint for Doxia sink/parser */
+        private final String roleHint;
+        private final String extension;
+        /** The name of the first element in case this is an XML format, otherwise {@code null} */
+        private final String firstElement;
+        private final boolean hasParser;
+        private final boolean hasSink;
 
-    private static final String CONFLUENCE_SINK = "confluence";
+        DoxiaFormat( String roleHint, String extension, boolean hasParser, boolean hasSink )
+        {
+            this( roleHint, extension, null, hasParser, hasSink );
+        }
 
-    private static final String DOCBOOK_SINK = "docbook";
+        DoxiaFormat( String roleHint, String extension, String firstElement, boolean hasParser, boolean hasSink )
+        {
+            this.roleHint = roleHint;
+            this.extension = extension;
+            this.firstElement = firstElement;
+            this.hasParser = hasParser;
+            this.hasSink = hasSink;
+        }
 
-    private static final String FO_SINK = "fo";
+        /**
+         * 
+         * @return the primary extension used with this format
+         */
+        public String getExtension()
+        {
+            return extension;
+        }
 
-    private static final String ITEXT_SINK = "itext";
+        public boolean hasParser()
+        {
+            return hasParser;
+        }
 
-    private static final String LATEX_SINK = "latex";
+        public boolean hasSink()
+        {
+            return hasSink;
+        }
 
-    private static final String RTF_SINK = "rtf";
+        /**
+         * 
+         * @return {@code true} in case this format is XML based
+         */
+        public boolean isXml()
+        {
+            return firstElement != null;
+        }
 
-    private static final String TWIKI_SINK = "twiki";
+        /**
+         * @param plexus not null
+         * @return an instance of <code>Parser</code> depending on the format.
+         * @throws ComponentLookupException if could not find the Parser for the given format.
+         * @throws IllegalArgumentException if any parameter is null
+         */
+        public Parser getParser( PlexusContainer plexus )
+            throws ComponentLookupException
+        {
+            if ( !hasParser )
+            {
+                throw new IllegalStateException( "The format " + this.toString() + " is not supported as parser!" );
```

Comment:
toString() is redundant.

Reply:
Fixed in https://github.com/apache/maven-doxia-converter/pull/38/commits/4fa08336ee9db0ae4d4eb4c22b68f8ebb22f6d7d.